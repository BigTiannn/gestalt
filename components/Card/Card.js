import React, { PropTypes } from 'react';
import classnames from 'classnames/bind';
import styles from './Card.css';

const cx = classnames.bind(styles);

export default function Card(props, context) {
    const { children } = props;
    return <div className={cx('Card')}>{children}</div>;
};

Card.propTypes = {
    children: PropTypes.node,
};
