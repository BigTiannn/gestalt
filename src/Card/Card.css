.card {
  composes: absolute from global;
  composes: rounded from global;
  background: rgba(0, 0, 0, 0.064);
  bottom: 0;
  left: 0;
  opacity: 0;
  right: 0;
  top: 0;
  z-index: -1;
}

.hover {
  animation: animateIn 0.2s cubic-bezier(0.31, 1, 0.34, 1) forwards;
}

.focus {
  opacity: 1;
  transform: scale(1.08, 1.06);
}

@keyframes animateIn {
  to {
    opacity: 1;
    transform: scale(1.08, 1.06);
  }
}
