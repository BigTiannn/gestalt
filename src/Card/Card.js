// @flow
import React, { Component, PropTypes } from 'react';
import Box from '../Box/Box';
import classnames from 'classnames';
import styles from './Card.css';

export default class Card extends Component {
  static PropTypes = {
    ariaLabel: PropTypes.string.isRequired,
    children: PropTypes.node
  }

  static props: {
    ariaLabel: String,
    children?: any
  }

  state = {
    focused: false,
    hovered: false
  };

  handleBlur = () => this.setState({ focused: false });
  handleFocus = () => { this.setState({ focused: true }); };
  handleMouseEnter = () => this.setState({ hovered: true });
  handleMouseLeave = () => this.setState({ hovered: false });

  render() {
    const {
      ariaLabel,
      children
    } = this.props;

    const {
      focused,
      hovered
    } = this.state;

    const classes = classnames(styles.card, {
      [styles.hover]: hovered,
      [styles.focus]: focused,
    });

    return (
      <Box
        aria-label={ariaLabel}
        onBlur={this.handleBlur}
        onFocus={this.handleFocus}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
        position="relative"
        tabIndex="0"
      >
        <div className={classes} />
        {children}
      </Box>
    );
  }
}
