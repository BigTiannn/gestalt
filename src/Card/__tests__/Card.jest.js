/* eslint-env jest */
/* eslint import/imports-first: 0 */
jest.unmock('../../Box/Box');
jest.unmock('../../Card/Card');

import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';

const snapshot = (component) => {
  const tree = renderer.create(
    component
  ).toJSON();
  expect(tree).toMatchSnapshot();
};

it('renders an empty Card', () => snapshot(<Card ariaLabel="Empty card" />));
it('renders a Card with text', () =>
  snapshot(<Card ariaLabel="Chris Lloyd - founder of Box">Chris Lloyd</Card>)
);
