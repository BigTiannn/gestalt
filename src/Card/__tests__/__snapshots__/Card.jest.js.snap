exports[`test renders a Card with text 1`] = `
<div
  aria-label="Chris Lloyd - founder of Box"
  className="box relative"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  tabIndex="0">
  <div
    className="card" />
  Chris Lloyd
</div>
`;

exports[`test renders an empty Card 1`] = `
<div
  aria-label="Empty card"
  className="box relative"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  tabIndex="0">
  <div
    className="card" />
</div>
`;
